module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,67 @@
+#include "compression.hpp"
+
+#include <iostream>
+#include <string>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, height>, width>& uncompressed) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(height * width);
+
+    for (const auto& row : uncompressed) {
+        auto rowIterator = row.cbegin();
+        uint8_t currentValue = *rowIterator;
+        uint8_t currentValueCounter = 1;
+        rowIterator++;
+
+        while (rowIterator != row.cend()) {
+            if (*rowIterator == currentValue) {
+                currentValueCounter++;
+            } else {
+                compressed.push_back(std::make_pair(currentValue, currentValueCounter));
+                currentValue = *rowIterator;
+                currentValueCounter = 1;
+            }
+            rowIterator++;
+        }
+        // and when row reach end:
+        compressed.push_back(std::make_pair(currentValue, currentValueCounter));
```

Comment:
More readable would be if you used only one **push_back()**, try to simplify it.